#### Задание 1.

Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

In [1]:
from math import factorial
import numpy as np

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

def binomial_p(p, n, k):
    return combinations(n, k) * pow(p, k) * pow((1 - p), (n - k))

In [2]:
p = 0.8
n = 100
k = 85
P = binomial_p(p, n, k)
print(P)

0.048061793700746355


Ответ: вероятность того, что стрелок из 100 раз попадет ровно 85 равна 4.81%

#### Задание 2.

Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

In [3]:
def puasson(lamb, m):
    return pow(lamb, m) * np.exp(-1 * lamb) / factorial(m)

In [4]:
p = 0.0004
n = 5000
lamb = n * p
m1 = 0
m2 = 2
P1 = puasson(lamb, m1)
P2 = puasson(lamb, m2)
print(P1, P2)

0.1353352832366127 0.2706705664732254


Ответ: а) вероятность, что ни одна лампочка не перегорит - 13.5%. б) что перегорит 2 - 27.1%

#### Задание 3.

Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [5]:
p = 0.5
n = 144
k = 70
P = binomial_p(p, n, k)
print(P)

0.06281178035144776


Ответ: вероятность выпадения орла ровно 70 раз при 144-х подбрасываниях равна 6.3%

#### Задание 4.

В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

In [6]:
# a)

p1 = combinations(7, 2) / combinations(10, 2)
p2 = combinations(9, 2) / combinations(11, 2)
p = p1 * p2
print(p1, p2, p)

0.4666666666666667 0.6545454545454545 0.3054545454545455


Вероятность того, что ровно 2 мяча белые считается как Р1 + Р2 + Р3, где

Р1 - вероятность того, что из первой корзины вытащили 2 белых мяча, а из второй 2 черных

Р2 - вероятность того, что из второй корзины вытащили 2 белых мяча, а из первой 2 черных

Р3 - вероятность того, что из первой и второй корзины вытащили по белому и черному мячу

P3 = P31 * P32,

где Р31 - вероятность вытащить белый и черный мяч из первой корзины,

Р32 - из второй

In [7]:
# b)

p1 = (combinations(7, 2) / combinations(10, 2)) * (combinations(2, 2) / combinations(11, 2))
p2 = (combinations(3, 2) / combinations(10, 2)) * (combinations(9, 2) / combinations(11, 2))
p31 = combinations(3, 1) * combinations(7, 1) / combinations(10, 2)
p32 = combinations(2, 1) * combinations(9, 1) / combinations(11, 2)
p3 = p31 * p32
p = p1 + p2 + p3
print(p1, p2, p3, p)

0.008484848484848484 0.04363636363636363 0.15272727272727274 0.20484848484848486


Вероятность того, что хотя бы один мяч белый можно посчитать следующим образом: P = 1 - Q, где Q - вероятность того, что все мячи черные

In [8]:
# c)

q = (combinations(3, 2) / combinations(10, 2)) * (combinations(2, 2) / combinations(11, 2))
p = 1 - q
print(p, q)

0.9987878787878788 0.0012121212121212121


Ответ:

а) вероятность того, что все мячи белые - 30.54%

б) вероятность того, что ровно два мяча белые - 20.48%

в) вероятность того, что хотя бы один мяч белый - 99.87%